compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/model/SimpleTypeField.java

```diff
@@ -32,16 +34,19 @@ public class SimpleTypeField {
     private final boolean isArrayType;
     private final NodeLocation nodeLocation;
     private final NodeLocation typeLocation;
+    private NodeList<AnnotationNode> annotations;
 
     public SimpleTypeField(String name, String type, boolean isValidType, boolean isNullable,
```

Comment:
This constructor is taking around 8 parameters. Wouldn't it be better if we could have a configuration record (Java record) which contains all the fields captured here and pass it as an argument to the constructor. I think it would make the code more readable.

Reply:
+1. we are planning to do code refactoring after update 09 as it is not extendable at the moment. We will consider this code improvement as well